client/connectivity/connectivity.go

```diff
@@ -198,8 +198,14 @@ func (c *circle) processPeers(ctx context.Context, peers []*pb.Peer) {
 }
 
 func (c *circle) newPeer(ctx context.Context, p *pb.Peer) *Peer {
+	isMe := p.GetName() == c.myName()
+
 	r := manual.NewBuilderWithScheme(fmt.Sprintf("rufs-%s", p.GetName()))
-	r.InitialState(peerToResolverState(p))
```

Comment:
This logic feels like it belongs inside of peerToResolverState. That would require passing some more things through (make peerToResolverState a method of circle, and probably have a reference from Peer to *circle), but it also means we have this logic in fewer places. WDYT?

Reply:
I indeed considered that and thought it would be less elegant if Peer had to have a reference back to circle as well. But perhaps that was my C++ lobe speaking, where references are bad. To compare, this is the patch when that reference is fine: https://github.com/sgielen/rufs/pull/68

Reply:
References aren't so bad in Go (though I don't actually know why they're bad in C++), it does mean the GC has to traverse it each time but that's fairly cheap given the small amount of peers we have. Circular references aren't a problem with Go's GC.

I like #68, it seems to have less code, in a better place.

Reply:
In C++, mostly because you have to be really careful that no reference ever becomes dangling. Especially in cycles that tends to become bug-prone pretty quickly. In Go, that's much less of a problem, of course. :)

I've merged #68, so will close this one :)